lib/signaling/v2/iceconnectionmonitor.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+const { ICE_ACTIVITY_CHECK_PERIOD_MS, ICE_INACTIVITY_THRESHOLD_MS } = require('../../util/constants');
+
+/**
+ * Monitors a {@link RTCPeerConnection}'s stats and notifies
+ * caller when inactivity is detected.
+ */
+class IceConnectionMonitor {
+  /**
+   * Construct an {@link IceConnectionMonitor}.
+   * @param {RTCPeerConnection} peerConnection
+   * @param {object} [options]
+   */
+  constructor(peerConnection, options) {
+    options = Object.assign({
+      activityCheckPeriodMs: ICE_ACTIVITY_CHECK_PERIOD_MS,
+      inactivityThresholdMs: ICE_INACTIVITY_THRESHOLD_MS,
+    }, options);
+
+    Object.defineProperties(this, {
+      _activityCheckPeriodMs: {
+        value: options.activityCheckPeriodMs
+      },
+      _inactivityThresholdMs: {
+        value: options.inactivityThresholdMs
+      },
+      _lastActiveStats: {
+        value: null,
+        writable: true
+      },
+      _peerConnection: {
+        value: peerConnection
+      },
+      _timer: {
+        value: null,
+        writable: true,
+      }
+    });
+  }
+
+  /**
+   * Get ICE connection stats, and extract received and send bytes.
+   * @returns Promise<?{timestamp, bytesReceived, bytesSent}>
+   */
+  _getIceConnectionStats() {
+    return this._peerConnection.getStats().then(stats => Array.from(stats.values()).find(stat => {
+      return stat.type === 'candidate-pair' && stat.nominated;
+    })).then((activeCandidatePairStats) => {
```

Comment:
@makarandp0 Nitpick: I think we can return the whole RTCIceCandidatePairStats here and get rid of the `then()` callback. Also, I think you can get rid of the braces around `activeIceCandidatePairStats`.

Reply:
imo having a limited fields returned from _getIceConnectionStats, helps with unit test/mocking of this function. I will  update  to get rid of braces.